conda/core/index.py

```diff
@@ -228,7 +342,12 @@ def push_spec(spec):
                 if ftr_name not in collected_track_features:
                     pending_track_features.add(ftr_name)
 
-    def push_record(record):
+    def push_record(record: Any) -> None:
```

Comment:
```suggestion
    def push_record(record: PackageRecord) -> None:
```

I'm pretty sure this argument is always meant to be a `PackageRecord`.